### PR TITLE
EVG-20078: Add merge queue options to project settings

### PR DESCRIPTION
**Pull request by minnakt**
EVG-20078

### Description
This PR adds radio button options to choose between the Evergreen and GitHub merge queue. This section is hidden behind the `isProduction()` flag which can be removed when the project is ready to be released.

### Screenshots
Despite the [LeafyGreen guidelines](https://www.mongodb.design/component/checkbox/guidelines/), I felt like radio buttons were more clear than a single checkbox. I think it presents the options in a way that's easier to understand. See also: the examples in [Don't Use Checkboxes](https://www.teamten.com/lawrence/programming/checkboxes/).

Screenshot 2023-07-03 at 10 56 53 AM

Screenshot 2023-07-03 at 11 42 49 AM

### Testing
- Updated `transformers.test.tsx`

### Evergreen PR
- evergreen-ci/evergreen/pull/6731 (must be merged and deployed first)

### CODE REVIEW

**Review by bsamek — APPROVED** [review]
design LGTM (someone from the UI team should review the code)

File: src/pages/projectSettings/tabs/GithubCommitQueueTab/getFormSchema.tsx [hunk 1]
```diff
@@ -213,6 +220,36 @@ export const getFormSchema = (
...
+                            title: "Evergreen",
+                            enum: [MergeQueue.Evergreen],
+                            description:
+                              "Use the standard commit queue owned and maintained by Evergreen.",
+                          },
+                          {
+                            type: "string" as "string",
+                            title: (
+                              <span>
+                                GitHub <BetaBadge>BETA</BetaBadge>
```
Comment (sophstad) [inline src/pages/projectSettings/tabs/GithubCommitQueueTab/getFormSchema.tsx:244]: [super nit] Probably preferable to apply `text-transform: uppercase` in the BetaBadge component.
```suggestion
                                GitHub <BetaBadge>Beta</BetaBadge>
```

**Review by sophstad — APPROVED** [review]
lgtm! (assuming tests pass once app is merged in)

**Comment by minnakt** [comment]
will merge this PR later as long as today's evergreen deploy goes well (= does not get reverted)

**Pull request merged**